en/docs/learn/username-recovery-via-user-preferred-notification-channel.md

```diff
@@ -0,0 +1,353 @@
+# Username Recovery Via User Preferred Notification Channel
+
+WSO2 Identity Server enables recovering user accounts where the user has forgotten the user name.
+
+!!! Note
+    This feature is only available via Account Recovery REST APis. Currently, WSO2 IS does not 
+    support this feature via the User Portal.    
+
+**Username Recovery Flow**
+
+ - User provides a set of unique claims to identify the user account.
+ - Then the server will prompt available notification channels to the user to receive username 
+ recovery notification.
+ - User selects a notification channel.
+ - Sever send the recovery notification to the user via the preferred notification channel.
+
+The following sections walk you through configuring and trying out username recovery via SMS and Email 
+Channels.   
+
+??? Warning "If you have migrated from a previous IS version"
+    If you have migrated from a previous IS version, ensure that the `IdentityMgtEventListener` with the
+    `         orderId=50        ` is set to **false** and that the Identity
+    Listeners with `         orderId=95        ` and
+    `         orderId=97        ` are set to **true** in the
+    `         <IS_HOME>/repository/conf/deployment.toml        ` file.
+   
+    !!! Note 
+        You can skip this step if there are no entries for `event.default_listener.xxx`, in 
+        the `deployment.toml` file.
+    
+    
+    ``` java
+    [event.default_listener.identity_mgt]
+    priority= "50"
+    enable = false
+    [event.default_listener.governance_identity_mgt]
+    priority= "95"
+    enable = true
+    [event.default_listener.governance_identity_store]
+    priority= "97"
+    enable = true
+    ```
+    
+## Configuring username recovery
+
+Following steps given below are to recover a user in the the super tenant, which is `carbon.super`.
```

Comment:
Follow the steps given below to recover a user in the super tenant (i.e., carbon.super).